lib/Driver/ToolChains.cpp

```diff
@@ -214,19 +210,26 @@ static void addCommonFrontendArgs(const ToolChain &TC,
   inputArgs.AddAllArgs(arguments, options::OPT_Xllvm);
   inputArgs.AddAllArgs(arguments, options::OPT_Xcc);
 
-  addOutputsOfType(arguments, output, inputArgs,
-                   types::TY_SwiftModuleDocFile,
-                   "-emit-module-doc-path");
-
   if (llvm::sys::Process::StandardErrHasColors())
     arguments.push_back("-color-diagnostics");
+}
 
+static void addModuleDocPathArgIfNecessary(const CommandOutput &output,
```

Comment:
Can you explain why this got factored out? It was already in a "common" location and now it gets invoked multiple times.

Reply:
The problem is that it should not run for a compile job with a supplementary output file map. The options that I can think of are to:
1. add a check where it was to avoid outputting it in that case,
2. pass in an extra parameter to the function where it used to be, or
3. yank it out of the add-common-arguments function and call it separately.

I chose #3, but I'm happy to revisit it. I'm not thrilled with any of the possibilities.

Reply:
How does this happen for all the other supplementary outputs? Or do we just…not have any besides modules and serialized diagnostics?

Reply:
I think these are the only two that were in the common routine. The others were inserted from compile-job-specific code. They were done via control flow, IIRC, so that's why I did it that way for these two.